src/test/compile-fail/never-fallback.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(never_type)]
```

Comment:
It's testing that the diverging type of `_` on line 30 defaults to `!` and not `()`. The file will build without `#![feaure(never_type)]` - it's the same as the run-pass/unit-fallback.rs test that I added except for that.